src/adapters/MercuryAdapterMock.ts

```diff
@@ -0,0 +1,50 @@
+import log from '../lib/log'
+import { MercuryAdapter } from '../MercuryAdapter'
+import {
+	OnHandleEvent,
+	OnConnectFunctionHandler,
+	OnErrorHandler,
+	IMercuryAdapterOnOptions,
+	IMercuryEventContract,
+	IMercuryEmitOptions
+} from '../types/mercuryEvents'
+
+export interface IMercuryAdapterMockOptions {}
+
+export default class MercuryAdapterMock<
+	EventContract extends IMercuryEventContract
+> implements MercuryAdapter<EventContract> {
+	public isConnected = true
+
+	public init(
+		options: IMercuryAdapterMockOptions,
+		_eventHandler: OnHandleEvent<IMercuryEventContract, any, any>,
+		_errorHandler: OnErrorHandler,
+		_onConnect: OnConnectFunctionHandler,
+		_onDisconnect: OnConnectFunctionHandler
+	): void {
+		// TODO
+		log.debug('MercuryAdapterMock.init()', { options })
+	}
+
+	public on<
+		EventName extends keyof EventContract,
+		EventSpace extends EventContract[EventName]
+	>(
+		_options: IMercuryAdapterOnOptions<EventContract, EventName, EventSpace>
+	): void {
```

Comment:
you could move caching in here and Mercury would not know the diff.

this would give much better tests of the mercury library vs testing the mock subclasses.